src/util/simplify_expr_int.cpp

```diff
@@ -1164,6 +1164,31 @@ bool simplify_exprt::simplify_bitnot(exprt &expr)
   return true;
 }
 
+/// simplifies string equalities as introduced in virtual function removal
+bool simplify_exprt::simplify_string_equality(exprt &expr)
+{
+  if(expr.id() != ID_equal)
+    return true;
+
+  const auto &eq = to_equal_expr(expr);
+  if(
+    eq.lhs().id() == ID_constant && eq.lhs().type().id() == ID_string &&
+    eq.rhs().id() == ID_constant && eq.rhs().type().id() == ID_string)
+  {
+    const auto &lhs_const = to_constant_expr(eq.lhs());
+    const auto &rhs_const = to_constant_expr(eq.rhs());
```

Comment:
I think this can be made to potentially catch more cases: just add an `ID_equal`/`ID_notequal` special case within `op0_is_const && op1_is_const` -- it should actually work for any type!